src/vsr/replica.zig

```diff
@@ -6706,14 +6706,36 @@ pub fn ReplicaType(
             return true;
         }
 
+        /// Primary must have no missing headers and no faulty prepares between op_repair_min
+        /// and self.op, to maintain the invariant that a replica can repair everything back
+        /// till op_repair_min
+        fn primary_journal_repaired(self: *Self) bool {
+            assert(self.status == .normal or self.status == .view_change);
+            assert(self.primary_index(self.view) == self.replica);
+            assert(self.view == self.log_view);
+
+            if (!self.valid_hash_chain_between(self.op_repair_min(), self.op)) return false;
+
+            for (self.op_repair_min()..self.op + 1) |op| {
+                const header = self.journal.header_with_op(op);
+                if (header == null or
```

Comment:
Hm, seems rather we can `assert(header != null)` here? Otherwise, we'd have bailed due to `!valid_hash_chain` before, no?

Reply:
Good point, addressed this in https://github.com/tigerbeetle/tigerbeetle/commit/6a2f725f5b883a9baa48e1a4f88cbabcc9288791!